### ISSUE TITLE
S3 download_file chunk_size can't be overridden

### ISSUE THREAD

**Issue by whitedr** [CargoSense/ex_aws#437]
I've just started playing around with ExAws (fantastic library!) and I think I ran across something which is a bug.  In testing out downloads from S3 via `ExAws.S3.download_file/4` I found that passing in `[max_concurrency: 8, chunk_size: 1024*1024*10, timeout: 120_000]` didn't change the default of making 1Mb chunk requests.  I think this may be due to this:

https://github.com/CargoSense/ex_aws/blob/master/lib/ex_aws/s3/download.ex#L30
and
https://github.com/CargoSense/ex_aws/blob/master/lib/ex_aws/s3/download.ex#L11

where it is looking for chunk_size in the Download `op` and not in `op.opts`.

**Comment by occurr** [comment]
If everyone gave this much information in their issues, the world would be a better place :)

Yes, you are 100% right. I will submit a pull request soon to fix this, as well as add the appropriate tests to catch this.

In the meantime you can work around this by updating your code to be: 

  ```
defstruct [
    :bucket,
    :path,
    :dest,
    opts: [],
    service: :s3,
    default_chunk_size: 1024 * 1024,
  ]
  ...
  def build_chunk_stream(op, config) do
    op.bucket
    |> get_file_size(op.path, config)
    |> chunk_stream(op.opts[:chunk_size] || op.default_chunk_size)
  end
````

**Comment by benwilson512** [comment]
Good catch, I'll handle this.

**Comment by benwilson512** [comment]
Fixed on master, will be releasing 1.1.3 shortly.
